doc/source/glossary.rst

```diff
@@ -344,199 +287,113 @@ Glossary
 
 
    Fortran order
-       See `column-major`
+       Same as `column-major <https://en.wikipedia.org/wiki/Row-_and_column-major_order>`_
 
 
    flattened
-       Collapsed to a one-dimensional array. See `numpy.ndarray.flatten`
-       for details.
+       See :term:`ravel`.
 
 
    homogeneous
-       Describes a block of memory comprised of blocks, each block comprised of
-       items and of the same size, and blocks are interpreted in exactly the
-       same way. In the simplest case each block contains a single item, for
-       instance int32 or float64.
+       All elements of a homogeneous array have the same type. ndarrays, in
+       contrast to Python lists, are homogeneous. The type can be complicated,
+       as in a :term:`structured array`, but all elements have that type.
 
-
-   immutable
-       An object that cannot be modified after execution is called
-       immutable.  Two common examples are strings and tuples.
+      NumPy `object arrays <#term-object-array>`_, which contain references to
+      Python objects, fill the role of heterogeneous arrays.
 
 
    itemsize
-       The size of the dtype element in bytes.
-
-
-   list
-       A Python container that can hold any number of objects or items.
-       The items do not have to be of the same type, and can even be
-       lists themselves::
-
-         >>> x = [2, 2.0, "two", [2, 2.0]]
-
-       The list `x` contains 4 items, each which can be accessed individually::
+        The size of the dtype element in bytes.
 
-         >>> x[2] # the string 'two'
-         'two'
 
-         >>> x[3] # a list, containing an integer 2 and a float 2.0
-         [2, 2.0]
-
-       It is also possible to select more than one item at a time,
-       using *slicing*::
-
-         >>> x[0:2] # or, equivalently, x[:2]
-         [2, 2.0]
-
-       In code, arrays are often conveniently expressed as nested lists::
-
-
-         >>> np.array([[1, 2], [3, 4]])
-         array([[1, 2],
-                [3, 4]])
-
-       For more information, read the section on lists in the `Python
-       tutorial <https://docs.python.org/tutorial/>`_.  For a mapping
-       type (key-value), see *dictionary*.
-
-
-   little-endian
-       When storing a multi-byte value in memory as a sequence of bytes, the
-       sequence addresses/sends/stores the least significant byte first (lowest
-       address) and the most significant byte last (highest address). Common in
-       x86 processors.
+   `little-endian <https://en.wikipedia.org/wiki/Endianness>`_
+       \
 
 
    mask
-       A boolean array, used to select only certain elements for an operation::
-
-         >>> x = np.arange(5)
-         >>> x
-         array([0, 1, 2, 3, 4])
-
-         >>> mask = (x > 2)
-         >>> mask
-         array([False, False, False, True,  True])
-
-         >>> x[mask] = -1
-         >>> x
-         array([ 0,  1,  2,  -1, -1])
+       The boolean array used to select elements in a :term:`masked array`.
```

Comment:
I think you need to give both meanings here. The old meaning was much more frequently needed than this new one.

Reply:
You're right. 511d769 restores old definition.